tests/src/test/fdc3.broadcast.ts

```diff
@@ -32,135 +30,155 @@ export default () =>
-        return new Promise(async (resolve) => {
+        const errorMessage = `\r\nSteps to reproduce:\r\n- Add fdc3.instrument context listener to app A\r\n- App A joins channel 1\r\n- App B joins channel 1\r\n- App B broadcasts fdc3.instrument context${documentation}`;
+
+        return new Promise(async (resolve, reject) => {
           //Add context listener to app A
-          listener = await window.fdc3.addContextListener(
-            "fdc3.instrument",
-            (context) => {
-              expect(context.type).to.be.equals("fdc3.instrument");
-              resolve();
-            }
-          );
+          listener = await window.fdc3.addContextListener(null, (context) => {
+            expect(context.type).to.be.equal("fdc3.instrument", errorMessage);
+            resolve();
+          });
 
-          assert.isObject(listener);
-          expect(typeof listener.unsubscribe).to.be.equals("function");
+          validateListenerObject(listener);
 
           //App A joins channel 1
           await joinChannel(1);
 
           //Open ChannelsApp app. ChannelsApp joins channel 1, then broadcasts context
           window.fdc3.open("ChannelsApp", channelsAppContext);
+
+          //if no context received throw error
+          await wait(3000);
+          reject(new Error(`${errorMessage} No context received`));
         });
       });
 
       it("Should receive context when joining a user channel then adding a context listener before app B broadcasts context to the same channel", async () => {
-        return new Promise(async (resolve) => {
-          const channels = await window.fdc3.getSystemChannels();
+        const errorMessage = `\r\nSteps to reproduce:\r\n- App A joins channel 1\r\n- Add listener of type fdc3.instrument to App A\r\n- App B joins channel 1\r\n- App B broadcasts fdc3.instrument context${documentation}`;
 
+        return new Promise(async (resolve, reject) => {
           //App A joins channel 1
           await joinChannel(1);
 
           //Add context listener to app A
-          listener = await window.fdc3.addContextListener(
-            "fdc3.instrument",
-            (context) => {
-              expect(context.type).to.be.equals("fdc3.instrument");
-              resolve();
-            }
-          );
+          listener = await window.fdc3.addContextListener(null, (context) => {
+            expect(context.type).to.be.equal("fdc3.instrument", errorMessage);
+            resolve();
+          });
 
-          assert.isObject(listener);
-          expect(typeof listener.unsubscribe).to.be.equals("function");
+          validateListenerObject(listener);
 
           //Open ChannelsApp app. channels app joins channel 1, then broadcasts context
           await window.fdc3.open("ChannelsApp", channelsAppContext);
+
+          //if no context received throw error
+          await wait(3000);
```

Comment:
3000 is a bit arbitrary, but I can see why it might be needed! Can you extract this as a constant, re-used across all tests?